src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/sink/CosmosDBSinkTask.java

```diff
@@ -52,30 +52,30 @@ public void put(Collection<SinkRecord> records) {
             return;
         }
 
-        logger.info("Sending " + records.size() + " records to be written");
+        logger.info("Sending {} records to be written", records.size());
 
         Map<String, List<SinkRecord>> recordsByContainer = records.stream()
                 //Find target collection for each record
                 .collect(Collectors.groupingBy(record ->
                         settings.getTopicContainerMap().getContainerForTopic(record.topic()).orElseThrow(
-                                () -> new IllegalStateException("No container defined for topic " + record.topic() + "."))));
+                                () -> new IllegalStateException(String.format("No container defined for topic %s .",
+                                record.topic())))));

```

Comment:
Why not use the `{}` notation here? Or does it not work for the record.topic()?

Reply:
as discussed, won't fix this.